MELA/python/mela_binding.cpp

```diff
@@ -0,0 +1,5145 @@
+#include "Mela.h"
+#include "TVar.hh"
+#include "TCouplingsBase.hh"
+#include "TMCFM.hh"
+#include "TLorentzVector.h"
+#include "pybind11/pybind11.h"
+#include "pybind11/stl.h"
+#include "pybind11/numpy.h"
+namespace py = pybind11;
+using namespace std;
+
+namespace pymela{
+    enum HQQ_indices{
```

Comment:
Until L215 and for later in other index duplications:

I think it might be better to simply name the enums in TCouplingBase.hh as
CouplingIndex_*, where * is consistent with SIZE_*, e.g.

```
enum CouplingIndex_HQQ{
    gHIGGS_KAPPA,
    gHIGGS_KAPPA_TILDE,

    SIZE_HQQ
};
```

You can then use

'using HQQ_indices = ::CouplingIndex_HQQ',

which would avoid duplicating the indices unnecessarily.

Now, this would not avoid the duplications downstream (value(...) statements). In order to do that, I think it might be better to convert each index set into a set of macros, e.g.

```
#define MELACMDS_INDEX_HQQ \
  MELACMD_INDEX_HQQ(gHIGGS_KAPPA) \
  MELACMD_INDEX_HQQ(gHIGGS_KAPPA_TILDE)
```

which allows you to write later something like

```
#define MELACMD_INDEX_HQQ(COUPLING) COUPLING,
enum CouplingIndex_HQQ{ MELACMDS_INDEX_HQQ SIZE_HQQ };
#undef MELACMD_INDEX_HQQ
...
#define MELACMD_INDEX_HQQ(COUPLING) .value(#COUPLING, pymela::COUPLING)
    py::enum_<pymela::HQQ_indices>(m, "HQQ_indices") MELACMDS_INDEX_HQQ .value("SIZE_HQQ", pymela::SIZE_HQQ);
#undef MELACMD_INDEX_HQQ
```

Reply:
This solution is quite welcome! I was looking around for something to prevent duplicating the code (I realized partway it was incredibly messy but  couldn't think of a clean solution) - so thank you. I'll do my best to implement this and clean up what I can.

Reply:
One question here - 

```#define MELACMDS_INDEX_HQQ \
  MELACMD_INDEX_HQQ(gHIGGS_KAPPA) \
  MELACMD_INDEX_HQQ(gHIGGS_KAPPA_TILDE)```

Is still duplicating code, albeit in a different way. One would still need to keep track of any/all couplings that could be added in the future. The fact that they are now defined as macros instead of explicitly as enumeration value arguments is now another layer of abstraction - I'm not sure this is the way to go.

I **will** look at using the macros to define the plethora of MELA couplings, however. That certainly seems useful.

Reply:
Well, you need to define the coupling somewhere.
```
#define MELACMDS_INDEX_HQQ \
  MELACMD_INDEX_HQQ(kappa, gHIGGS_KAPPA) \
  MELACMD_INDEX_HQQ(kappa_tilde, gHIGGS_KAPPA_TILDE)
  
#define MELACMD_INDEX_HQQ(COUPLINGSTR, COUPLING) COUPLING,
enum CouplingIndex_HQQ{ MELACMDS_INDEX_HQQ SIZE_HQQ };
#undef MELACMD_INDEX_HQQ
```
should be sufficient to declare the enums once. For the rest, you can use `COUPLINGSTR` or `COUPLING`, or both, within the `#define`-`#undef` blocks of `MELACMD_INDEX_HQQ`.

Reply:
I've simply kept the enums as is, but now pull directly from the named TCouplingsBase indices